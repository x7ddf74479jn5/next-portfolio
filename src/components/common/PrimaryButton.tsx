import styles from "src/styles/components/common/PrimaryButton.module.scss";

import type { ButtonType } from "../../types/utils/index";

type Props = ButtonType & {
  label?: string;
};

const PrimaryButton: React.FC<Props> = ({ children, disabled = false, label, ...rest }) => {
  return (
    <button className={styles.btnRound} disabled={disabled} {...rest}>
      {children || label}
    </button>
  );
};

export default PrimaryButton;
