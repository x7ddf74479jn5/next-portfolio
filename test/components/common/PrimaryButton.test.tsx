import { cleanup, fireEvent, render } from "@testing-library/react";
import PrimaryButton from "src/components/common/PrimaryButton";

afterEach(cleanup);

describe("PrimaryButton", () => {
  const buttonLabel = "hoge";

  it("matches snapshot", () => {
    const { asFragment } = render(<PrimaryButton label={buttonLabel} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("has text by label", () => {
    const { container } = render(<PrimaryButton label={buttonLabel} />, {});
    expect(container).toHaveTextContent(buttonLabel);
  });

  it("has text by children", () => {
    const { container } = render(<PrimaryButton>{buttonLabel}</PrimaryButton>);
    expect(container).toHaveTextContent(buttonLabel);
  });

  it("onClick", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <PrimaryButton
        onClick={() => {
          return handleClick();
        }}
      >
        {buttonLabel}
      </PrimaryButton>
    );
    fireEvent.click(getByText(buttonLabel));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
  it("disabled", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <PrimaryButton
        onClick={() => {
          return handleClick();
        }}
        disabled
      >
        {buttonLabel}
      </PrimaryButton>
    );
    expect(getByText(buttonLabel)).toBeDisabled;
    fireEvent.click(getByText(buttonLabel));
    expect(handleClick).toHaveBeenCalledTimes(0);
  });
});
