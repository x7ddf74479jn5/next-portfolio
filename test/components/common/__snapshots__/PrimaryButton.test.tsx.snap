// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="btnRound"
  >
    hoge
  </button>
</DocumentFragment>
`;
